pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -260,12 +317,122 @@ func zipFiles(files []string, zipFileName string) error {
+// anonymizer, then it will create the anonymized name and store it in
+// the anonymizer.
+func getPodName(podName string, anon bool, anonymizer *anonymizer) string {
+	// If we aren't anonymizing the logs, just return the podName
+	if !anon {
+		return podName
+	}
+
+	// If this pod name has already been mapped, return that
+	if anonName, ok := anonymizer.podNames[podName]; ok {
+		return anonName
+	}
+
+	// the podName hasn't been anonymized yet so we split it up
+	// so we can anonymize the namespace
+	nameComponents := strings.SplitN(podName, ".", 2)
+	if len(nameComponents) != 2 {
+		unknownPodName := "anonPod.anonNamespace"
+		anonymizer.podNames[podName] = unknownPodName
+		return unknownPodName
+	}
+	var anonPodName, anonNamespace string
+	name, namespace := nameComponents[0], nameComponents[1]
+	if val, ok := anonymizer.namespaces[namespace]; ok {
+		anonNamespace = val
+	} else {
+		anonNamespace = fmt.Sprintf("namespace-%d", len(anonymizer.namespaces)+1)
+		anonymizer.namespaces[namespace] = anonNamespace
+	}
+
+	// we want to special case the traffic-manager so we can easily distinguish
+	// between that and the traffic-agents
+	if strings.Contains(name, "traffic-manager") {
+		anonPodName = fmt.Sprintf("traffic-manager.%s", anonNamespace)
+	} else {
+		anonPodName = fmt.Sprintf("pod-%d.%s", len(anonymizer.podNames)+1, anonNamespace)
+	}
+	// Store the anonPodName in the map
+	anonymizer.podNames[podName] = anonPodName
+	return anonPodName
+}
+
+// anonymizeLog is a helper function that replaces the namespace + podName
+// used in the log with its anonymized version, provided by the anonymizer.
+// It overwrites the file with the anonymized version.
+func anonymizeLog(stdout io.Writer, logFile string, anonymizer *anonymizer) error {
+	// Read the contents we are going to overwrite from the file
+	content, err := os.ReadFile(logFile)
+	if err != nil {
+		return err
+	}
+	// Open the file with write so we can overwrite it
+	stringContent := string(content)
+	f, err := os.OpenFile(logFile, os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	// First we replace the actual namespace with the anonymized
+	// version.
+	for namespace, anonNamespace := range anonymizer.namespaces {
+		stringContent = strings.ReplaceAll(stringContent, namespace, anonNamespace)
+	}
+	// Now we do pod name which is a little bit more complicated
+	for fullPodName, fullAnonPodName := range anonymizer.podNames {
+		// strip the namespace off of the anonymized name
+		anonPodParts := strings.Split(fullAnonPodName, ".")
+		anonPodName := anonPodParts[0]
+
+		// Strip the namespace off of the podName
+		podParts := strings.Split(fullPodName, ".")
+
+		// replace pod name + full hash with the anonymized name
+		podName := podParts[0]
+		stringContent = strings.ReplaceAll(stringContent, podName, anonPodName)
+
+		// Strip the hash off of the pod name + replace remainder with anonymized name
```

Comment:
wait im confused why do we do both of these things? it seems either/or would be sufficient?

Reply:
Ah I should better describe this in the logs.. So the problem was in the traffic-manager logs we might have something like this:
```
2021-10-01 18:45:36.8322 info    agent-injector : Injecting traffic-agent into pod echo-auto-inject-869ff95bc7-      .default
```
but in the traffic-agent we have logs like this:
```
2021-10-01 18:45:38.3415 info    {Name:echo-auto-inject Namespace:default PodIP:10.42.0.35 AgentPort:9900 AppMo      unts:/tel_app_mounts AppPort:8080 ManagerHost:traffic-manager.ambassador ManagerPort:8081}
```

So I could just ReplaceAll without the hash, but then the first log would look like:
```
2021-10-01 18:45:36.8322 info    agent-injector : Injecting traffic-agent into pod pod-1-869ff95bc7-.namespace-2:
```

Although I just noticed my code wouldn't anonymize this bc it doesn't include the 2nd part of the hash...

Hm I think I have an idea that might remedy this + the issue with replicasets + statefulsets